components/uniset/src/props.rs

```diff
@@ -0,0 +1,297 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+#![allow(clippy::unreadable_literal, dead_code)]
+
+use crate::provider::*;
+use crate::UnicodeSet;
+use icu_provider::prelude::*;
+use std::borrow::Cow;
+use std::convert::TryInto;
+
+// helper fn
+fn get_prop<'d, D: DataProvider<'d, UnicodeProperty<'d>> + ?Sized>(
+    ppucd_provider: &D,
+    resc_key: ResourceKey,
+) -> UnicodeSet {
+    let data_req = DataRequest {
+        resource_path: ResourcePath {
+            key: resc_key,
+            options: ResourceOptions {
+                variant: None,
+                langid: None,
+            },
+        },
+    };
+    let mut resp: DataResponse<UnicodeProperty> = ppucd_provider.load_payload(&data_req).unwrap();
```

Comment:
Use `?` instead of `.unwrap()` and make these functions return `Result<UnicodeSet, UnicodeSetError>`.

Reply:
Done.